src/d3a/models/strategy/area_agents/one_sided_engine.py

```diff
@@ -220,14 +225,14 @@ def event_offer_changed(self, *, market_id, existing_offer, new_offer):
             self.offer_age[new_offer.id] = self.offer_age.pop(existing_offer.id)
             offer_info = self.forwarded_offers[existing_offer.id]
             forwarded = self._forward_offer(new_offer, new_offer.id)
-
-            self.owner.log.info("Offer %s changed to residual offer %s",
-                                offer_info.target_offer,
-                                forwarded)
-
-            # Do not delete the forwarded offer entries for the case of residual offers
-            if existing_offer.seller != new_offer.seller:
-                self._delete_forwarded_offer_entries(offer_info.source_offer)
+            if forwarded:
```

Comment:
In order to avoid nesting, you can early-return here by using : 
if not forwarded: return

Reply:
ok, will do!